frontend/src/components/App.tsx

```diff
@@ -44,3 +86,25 @@ function App() {
 }
 
 export default App;
+
+function RequireAuth({ children }: { children: JSX.Element }) {
+  const [cookies, setCookies] = useCookies();
+  let location = useLocation();
+
+  if (!cookies.user) {
+    return <Navigate to="/signin" state={{ from: location }} replace />;
+  }
+
+  return children;
+}
+
+function RequireGuest({ children }: { children: JSX.Element }) {
+  const [cookies, setCookies] = useCookies();
+  let location = useLocation();
+
+  if (cookies.user) {
+    return <Navigate to="/" state={{ from: location }} replace />;
+  }
+
+  return children;
+}
```

Comment:
Create a middleware for this one to reduce the code duplication in the routes.

Reply:
i put these on separate files, and exported them there